src/storage/types.rs

```diff
@@ -109,6 +109,9 @@ pub struct Fingerprint {
 pub struct CardNumber(masking::StrongSecret<String>);
 
 impl CardNumber {
+    pub fn inner(&self) -> &masking::StrongSecret<String> {
+        &self.0
+    }
```

Comment:
Let's not have this function, instead, implement validate trait for this as well, and call them as a chain.